autotest/utilities/test_gdalalg_raster_edit.py

```diff
@@ -261,6 +307,48 @@ def test_gdalalg_raster_pipeline_edit_bbox(tmp_vsimem):
         assert ds.GetGeoTransform() == pytest.approx((1.0, 0.45, 0.0, 200.0, 0.0, -9.9))
 
 
+def test_gdalalg_raster_pipeline_edit_nodata(tmp_vsimem):
+
+    out_filename = str(tmp_vsimem / "tmp.tif")
+    gdal.FileFromMemBuffer(out_filename, open("../gcore/data/byte.tif", "rb").read())
+
+    pipeline = get_pipeline_alg()
+    assert pipeline.ParseRunAndFinalize(
+        [
+            "read",
+            "../gcore/data/byte.tif",
+            "!",
+            "edit",
+            "--nodata=100",
+            "!",
+            "write",
+            "--overwrite",
+            out_filename,
+        ]
+    )
+
```

Comment:
Should we test also invalid/out of range nodata values, like -100 or 300 for byte?

Reply:
check added